api/v1beta1/metal3cluster_webhook.go

```diff
@@ -68,6 +74,79 @@ func (c *Metal3Cluster) validate() error {
 		)
 	}
 
+	if c.Spec.CloudProviderEnabled != nil && c.Spec.NoCloudProvider != nil {
+		if !*c.Spec.CloudProviderEnabled && !*oldM3C.Spec.NoCloudProvider {
```

Comment:
We can merge these two to `if *c.Spec.CloudProviderEnabled == *oldM3C.Spec.NoCloudProvider`, and what if `oldM3C == nil`?

Reply:
oh, this is a mistake it should be 
`if !*c.Spec.CloudProviderEnabled && !*c.Spec.NoCloudProvider {`
Let me fix

Reply:
fixed